sql/delete.go

```diff
@@ -63,14 +66,35 @@ func (p *planner) Delete(n *parser.Delete) (planNode, error) {
 		if err := node.Err(); err != nil {
 			return nil, err
 		}
+		values := node.Values()
 
-		// TODO(tamird/pmattis): delete the secondary indexes too
-		primaryKey, err := encodeIndexKey(index, colMap, node.Values(), indexKey)
+		primaryIndexKeySuffix, err := encodeIndexKey(primaryIndex, colMap, values, nil)
 		if err != nil {
 			return nil, err
 		}
-		rowStartKey := proto.Key(primaryKey)
-		b.DelRange(rowStartKey, rowStartKey.PrefixEnd())
+		primaryIndexKey := bytes.Join([][]byte{primaryIndexKeyPrefix, primaryIndexKeySuffix}, nil)
+
+		rowStartKey := proto.Key(primaryIndexKey)
+		rowEndKey := rowStartKey.PrefixEnd()
+		if log.V(2) {
+			log.Infof("DelRange %q - %q", rowStartKey, rowEndKey)
+		}
+		b.DelRange(rowStartKey, rowEndKey)
+
+		// Delete the secondary indexes.
+		secondaryIndexKeys, err := encodeSecondaryIndexKeys(tableDesc, colMap, values, primaryIndexKeySuffix)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, secondaryIndexKey := range secondaryIndexKeys {
+			// TODO:(tamird): check uniqueness constraints (primary and
+			// secondary). Probably will have to do some scans?
```

Comment:
Why do you have to check uniqueness constraints on delete? I would think that is only necessary on INSERT and UPDATE.